src/mpi/coll/bcast.c

```diff
@@ -1567,7 +1567,7 @@ int MPI_Bcast( void *buffer, int count, MPI_Datatype datatype, int root,
     /* ... body of routine ...  */
     
     mpi_errno = MPID_Bcast( buffer, count, datatype, root, comm_ptr, &errflag );
-    if (mpi_errno) goto fn_fail;
+    if (mpi_errno) MPIR_ERR_POP(mpi_errno);
```

Comment:
MPIR_ERR_POP is a wrapper for "goto fn_fail" with additional capabilities to handle errors. In case when HAVE_ERROR_CHECKING is defined MPIR_ERR_POP calls MPIR_Err_create_code(...) and then "goto fn_fail". If HAVE_ERROR_CHECKING is not defined then MPIR_ERR_POP is just "goto fn_fail".